.github/workflows/ci.yml

```
@@ -0,0 +1,459 @@
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+        options: --health-cmd="pg_isready" --health-interval=10s --health-timeout=5s --health-retries=5
+        ports:
+          - 5432:5432
+
+    env:
+      artifacts_name: >-
+        PHP ${{ matrix.php }}
+        ${{ matrix.composer_args == '--prefer-lowest' && 'prf-low' || '' }}
+        ${{ matrix.db }}
+        ${{ matrix.phpunit == 'true' && 'phpunit' || '' }}
+        ${{ matrix.phpunit == 'true' && matrix.phpunit_suite || '' }}
+        ${{ matrix.endtoend == 'true' && 'endtoend' || '' }}
+        ${{ matrix.endtoend == 'true' && matrix.endtoend_suite || '' }}
+        ${{ matrix.js == 'true' && 'js' || '' }}
+        ${{ matrix.phpcoverage == 'true' && 'phpcoverage' || '' }}
+        ${{ matrix.phplinting == 'true' && 'phplinting' || '' }}
+        ${{ matrix.name_suffix }}
+
+    name: >-
+      PHP ${{ matrix.php }}
+      ${{ matrix.composer_args == '--prefer-lowest' && 'prf-low' || '' }}
+      ${{ matrix.db }}
+      ${{ matrix.phpunit == 'true' && 'phpunit' || '' }}
+      ${{ matrix.phpunit == 'true' && matrix.phpunit_suite || '' }}
+      ${{ matrix.endtoend == 'true' && 'endtoend' || '' }}
+      ${{ matrix.endtoend == 'true' && matrix.endtoend_suite || '' }}
+      ${{ matrix.js == 'true' && 'js' || '' }}
+      ${{ matrix.phpcoverage == 'true' && 'phpcoverage' || '' }}
+      ${{ matrix.phplinting == 'true' && 'phplinting' || '' }}
+      ${{ matrix.name_suffix }}
+
+    steps:
+
+      - name: Checkout code
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # @v2
+
+      - name: Install PHP
+        # SHA will need to be updated to support new php version when they are released
+        uses: shivammathur/setup-php@6cba851606e69e00775ebccba68bc1d418b6fa44 # @v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: curl, dom, gd, intl, json, ldap, mbstring, mysql, tidy, xdebug, zip
+          tools: composer:v2
+          coverage: xdebug
+          # While this should be the correct way to allow forks in composer.json repositories
+          # in practice there are still many sporadic "Could not authenticate against github.com" errors
+          # there's 1,000 requests per hour limit when using this token, likely it get exceeded
+          # fairly easily when using a fork with multiple jobs in a matrix
+          #env:
+          # COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Configure PHP
+        run: |
+          # Set memory limit and disable xdebug if not running phpcoverage
+          if [ -z $(which php) ]; then
+            echo "PHP not installed, skipping" && exit 0
+          fi
+
+          # github linux runners have 7GB of RAM
+          # https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
+          # Set a high memory limit, particularly for php coverage tests
+          PHP_MEMORY_LIMIT=6144M
```

```suggestion
          PHP_MEMORY_LIMIT=6G
```
More immediately clear it's exactly 6GB without having to divide by 1024 to confirm.

---

Updated